Publish minutes of 2022-01-20 meeting (#149)
hober/web-extensions: _minutes/2022-01-20-wecg.md

## Attendees (sign yourself in)

 1. Ales Pospichal (I.CA)
 2. Alexei (Privacy Badger)
 3. Rob Wu (Mozilla)
 4. Giorgio Maone (NoScript)
 5. Nick McGuire (1Password)
 6. Carlos Jeurissen (Jeurissen Apps)
 7. Philipp Claßen (Ghostery)
 8. Simeon Vincent (Google)
 9. Oliver Dunk (1Password)
 10. Krzysztof Modras (Ghostery)
 11. Mukul Purohit (Microsoft)
 12. Richard Worth (Capital One)
 13. Emilia Paz (Google)
 14. Bradley Cushing (Dashlane)
 15. Nir Nahum (WalkMe)
 16. Jessie Berlin (Apple)
 17. Tomislav Jovanovic (Mozilla)
 18. Mélanie Chauvel (Dashlane)

## Meeting notes

[Issue 145](https://github.com/w3c/webextensions/issues/145): Proposals for improving WECG meetings

 * [simeon] Introduced the topic in the last meeting, and elaborated in issue 145. Not that much to discuss today; encourage everyone to take a look and offer feedback.
 * [tomislav] Timeboxing ([issue 144](https://github.com/w3c/webextensions/issues/144)) is a way to run a meeting. That suggestion isn't very controversial.
 * [simeon] Idea of “champions” is just thinking aloud loud. Encourage everyone to take a look at the “Keeping meetings productive” section.

>  * [tomislav] Discussions that can be asynchronous/offline (i.e. in issues on Github) can better be discussed asynchronously.

Reviewer: ```suggestion
 * [tomislav] Discussions that can be asynchronous/offline (i.e. in issues on Github) can better be discussed asynchronously.
 * [jessie] The "champions" suggestion can be very effective for driving a topic or project to conclusion.
```

I tried to follow up with Jessie over email to make sure she's comfortable with this edit, but I haven't heard back. @xeenon, might you be able to check in?

[Issue 146](https://github.com/w3c/webextensions/issues/146): Clarify browser vendor positions on APIs in MV3

 * [oliver] Is storage.session API going to be adopted by other browsers? Saw comments about structured cloning, can that be used to store non-JSON data?
 * [tomislav] API is going to be the same. Structured cloning may be an option.
 * [rob] We haven't started on implementation yet, but broadly it makes sense to have a consistent object serialization format across storage apis
 * [jessie] (on behalf of Apple) Policy to not talk about unreleased APIs. The currently released MV3 API in Safari doesn't have storage.session.
 * [oliver] cross-extension messaging: will runtime.sendMessage to message other extensions continue to be supported?
 * [simeon] On Chrome this API should continue to work as expected. If it doesn't, that's a bug and we should fix it.
 * [rob] On Firefox side, no plans to remove it either.
 * [jessie] In Safari this API is not implemented.

>  * [simeon] Let's consider a process for aligning on extension APIs, e.g. using labels on Github.

Reviewer: ```suggestion
 * [simeon] Let's consider a process for more clearly communicating positions on extension APIs, e.g. using labels on Github.
```

 * [rob] So far the decisions were captured in meeting notes, and in some cases repeated in comments on Github.

[Issue 120](https://github.com/w3c/webextensions/issues/120): Manifest V3 vs native messaging [Ales Pospichal]

 * [ales] Final decision / next steps on service worker limit? Native messaging app used by extension (smartcard). The app supports a content script via the background, and getting killed after 5 minutes interferes with the extension's functionality.
 * [simeon] There are two relevant issues here in Chrome. Native messaging host terminated after 5 minutes, Native messaging host terminated too. Behavior not intentional and a high-prio issue.
   * https://bugs.chromium.org/p/chromium/issues/detail?id=1152255

>    * https://bugs.chromium.org/p/chromium/issues/detail?id=1030305

Reviewer: ```suggestion
 * [simeon] There are two relevant issues here in Chrome: service worker terminated after 5 minutes (crbug.com/1152255) and native messaging host not being terminated on SW disconnect (crbug.com/1030305). Neither is intentional. Making it possible for extension SWs to live longer than 5 is one of our top priorities. 
```

 * [tomislav] Firefox does not intend to support the 5 minute limit that Chrome has.
 * [bradley] We've tested the Safari preview, and cannot use non-blocking webRequest in its background script. What's the future of non-blocking webRequest across browsers?
 * [jessie] Generally, try to avoid keeping the extension alive indefinitely. Would like to hear more about the use cases.
 * [krysztof] There are existing examples on the issue tracker, e.g. [issue 72](https://github.com/w3c/webextensions/issues/72), [issue 88](https://github.com/w3c/webextensions/issues/88). The general problem with the declarative API is that advertising technology evolves rapidly and privacy/security extensions will not be able to keep up with declarative API alone.

>  * [simeon] Keeping service worker alive to observe network activity async is not ideal, but a currently acceptable middle-ground between constantly keeping a SW alive and totally removing webRequest.

Reviewer: ```suggestion
 * [simeon] Keeping service worker alive to observe asynchronously observe network activity is not ideal, but during the design process for MV3 we considered it acceptable middle-ground between constantly keeping the background context alive and totally removing webRequest. 
```

[Issue 115](https://github.com/w3c/webextensions/issues/115): move browser specific settings to browser_specific_settings key

 * [carlos] Let other browsers align on Firefox/Edge's way of specifying metadata such as minimum version.
 * [simeon] Initial feedback from Chrome engineering is hesitation - we ignore unknown keys and could break existing extensions if we were to relocate properties. If the value of this feature is more clear we can reconsider if needed.
 * [tomislav] We feel this is useful both for browser vendors and developers. For example, if we decide to support a feature similar to Chrome, but with a different shape, we either have to use another manifest key, or developers have to use separate manifests for different browsers.  In general, IMO top level keys are a good match for something that is agreed between browsers, and browser_specific_settings could be used for experimental things (until a final design is agreed between browsers) or for purposefully different things like min/max supported browser versions (which will forever be browser-specific).

> [Issue 147](https://github.com/w3c/webextensions/issues/147): Inconsistency: determine when non-persistent background pages get suspended()

Reviewer: ```suggestion

[Issue 147](https://github.com/w3c/webextensions/issues/147): Inconsistency: determine when non-persistent background pages get suspended()
```

At the moment everything form Issue 115 to Jessie's section is being very spaced out (each &lt;li> tag contains a &lt;p> tag). Inserting a new line above this line fixes the problem.

 * [carlos] Had previously raised this in the past. Created an issue because I believe this is something the group should do.
 * [tomislav] The details are very implementation-specific, not sure if this is something that can be specified.
 * [rob] Are we looking to clarify current behavior or attempt to standardize on a specific set of behaviors?

>  * [alexei] If everyone were to implement Mozilla's Limited Event page proposal ([issue 134](https://github.com/w3c/webextensions/issues/134)), then we would not have this inconsistency.

Reviewer: ```suggestion
 * [alexei] If everyone were to implement Mozilla's Limited Event page proposal ([issue 134](https://github.com/w3c/webextensions/issues/134)), then we would not have this inconsistency. Still have not heard a satisfactory reason why Google opposes this.
```

@ghostwords does this tweak look good to you?

Reviewer 2: Yes, thank you!

I think the closest to an actual explanation thus far has been that [Google cares about low-end devices](https://github.com/w3c/webextensions/blob/main/_minutes/2021-12-09-wecg.md#:~:text=Especially%20a%20constraint%20for%20low%20end%20devices.), which, given that [SWs will probably use more not fewer system resources](https://www.eff.org/deeplinks/2021/12/googles-manifest-v3-still-hurts-privacy-security-innovation#:~:text=Google%E2%80%99s%20performance%20claims) overall, does not make any sense.

>  * [simeon] Objective is to minimize lifetime of pseudo-persistent background scripts. Concern with “event pages” is that developers may incorrectly assume that the current behavior of Chrome's MV2 event page is going to be supported. Currently working on getting our proposal into a publicly shareable state. Hoping to have shared in time for the next public session.

Reviewer: ```suggestion
 * [simeon] Objective is to minimize lifetime of pseudo-persistent background scripts. Concern with “event pages” is that developers may incorrectly assume that the current behavior of Chrome's MV2 event page is going to be supported. Currently working on getting our proposal into a publicly shareable state. Hoping to have shared in time for the next public session. Comparing these proposals may help clarify our position better than I've been able to.
```
